src/CSharp/CodeCracker/Style/TaskNameAsyncAnalyzer.cs

```diff
@@ -0,0 +1,54 @@
+﻿using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using System.Collections.Immutable;
+using System.Threading.Tasks;
+
+namespace CodeCracker.Style
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class TaskNameAsyncAnalyzer : DiagnosticAnalyzer
+    {
+        public const string DiagnosticId = "CC0061";
+        internal const string Title = "Async method can be terminating with 'Async' name.";
+        internal const string MessageFormat = "Change method name including 'Async'.";
+        internal const string Category = SupportedCategories.Style;
+        const string Description = "Async method can be terminating with 'Async' name.";
+
+        internal static DiagnosticDescriptor Rule = new DiagnosticDescriptor(
+            DiagnosticId,
+            Title,
+            MessageFormat,
+            Category,
+            DiagnosticSeverity.Info,
+            isEnabledByDefault: true,
+            description: Description,
+            helpLink: HelpLink.ForDiagnostic(DiagnosticId));
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get { return ImmutableArray.Create(Rule); } }
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.RegisterSyntaxNodeAction(AnalyzeClass, SyntaxKind.MethodDeclaration);
+        }
+
+        public Task<bool> xpto()
+        {
+            return null;
+        }
+
+        private static void AnalyzeClass(SyntaxNodeAnalysisContext context)
+        {
+            var invocationExpression = (MethodDeclarationSyntax)context.Node;
+
+            var nameSyntax = invocationExpression.ReturnType?.ToString();
+            if (nameSyntax == null || !nameSyntax.Contains("Task")) return;
```

Comment:
Not the modifier, @carloscds, because an async method might not have the `async` modifier.
But consider this:

````csharp
class MyPromise : Task {}
class Foo
{
    async MyPromise DoSomething()
    {
    }
}
````

This method should actually be called `DoSomethingAsync`, but currently it won't be detected, as the return type is called `MyPromise`, and does not have `Task` in its name.
You have to check if the return type is a `Task`, or if it inherits from `Task`.

Check this method I added, `AllBaseTypesAndSelf`:
https://github.com/giggio/code-cracker/blob/unused-parameters/src/CSharp/CodeCracker/Extensions/AnalyzerExtensions.cs#L152

See also how it is used. Is it not in yet, as we are still discussing that PR, but that is the idea.